xbmc/video/VideoDatabase.cpp

```diff
@@ -11408,3 +11554,564 @@ void CVideoDatabase::EraseAllForPath(const std::string& path)
+                                        VideoVersionItemType& itemType)
+{
+  idFile = GetFileId(strFilenameAndPath);
+  if (idFile < 0)
+    return -1;
+
+  return GetVideoVersionInfo(idFile, strVideoVersion, idMedia, mediaType, itemType);
+}
+
+int CVideoDatabase::GetVideoVersionInfo(int idFile,
+                                        std::string& strVideoVersion,
+                                        int& idMedia,
+                                        std::string& mediaType,
+                                        VideoVersionItemType& itemType)
+{
+  if (!m_pDB || !m_pDS)
+    return -1;
+
+  int idVideoVersion = -1;
+
+  try
+  {
+    m_pDS->query(PrepareSQL("SELECT videoversiontype.name AS name,"
+                            "  videoversiontype.id AS id,"
+                            "  videoversion.idMedia AS idMedia,"
+                            "  videoversion.mediaType AS mediaType,"
+                            "  videoversion.itemType AS itemType "
+                            "FROM videoversion"
+                            "  JOIN videoversiontype ON "
+                            "    videoversiontype.id = videoversion.idType "
+                            "WHERE videoversion.idFile = %i", idFile));
+
+    if (m_pDS->num_rows() > 0)
+    {
+      idVideoVersion = m_pDS->fv("id").get_asInt();
+      strVideoVersion = m_pDS->fv("name").get_asString();
+      idMedia = m_pDS->fv("idMedia").get_asInt();
+      mediaType = m_pDS->fv("mediaType").get_asString();
+      itemType = static_cast<VideoVersionItemType>(m_pDS->fv("itemType").get_asInt());
+    }
+
+    m_pDS->close();
+  }
+  catch (...)
+  {
+    CLog::Log(LOGERROR, "{} failed for {}", __FUNCTION__, idFile);
+  }
+
+  return idVideoVersion;
+}
+
+bool CVideoDatabase::GetVideoVersionsNav(const std::string& strBaseDir,
+                                         CFileItemList& items,
+                                         VideoDbContentType idContent /* = UNKNOWN */,
+                                         const Filter& filter /* = Filter() */,
+                                         int idMedia /* = -1 */)
+{
+  if (!m_pDB || !m_pDS)
+    return false;
+
+  std::string type;
+  if (idContent == VideoDbContentType::MOVIES)
+  {
+    type = MediaTypeMovie;
+  }
+  else
+    return false;
+
+  CVideoDbUrl videoUrl;
+  if (!videoUrl.FromString(strBaseDir))
+    return false;
+
+  try
+  {
+    std::string strSQL;
+
+    if (idMedia != -1)
+      strSQL = PrepareSQL("SELECT videoversiontype.name AS name,"
+                          "  videoversiontype.id AS id "
+                          "FROM videoversiontype"
+                          "  JOIN videoversion ON"
+                          "    videoversion.idType = videoversiontype.id "
+                          "WHERE idMedia = %i AND mediaType = '%s'", idMedia, type.c_str());
+    else
+      strSQL = PrepareSQL("SELECT videoversiontype.name AS name,"
```

Comment:
This needs to be `SELECT DISTINCT ...` otherwise if you add 10 films to say 4K HDR, the nav will list 4K HDR 10 times and each one will contain the 10 films.

Reply:
I've fixed this, you can't add duplicate versions.